hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/processors/ProcessorModule.java

```diff
@@ -53,7 +56,10 @@ static ContractValidationRule providePrefixCodeRule() {
 
     @Provides
     @Singleton
-    static Map<Address, PrecompiledContract> provideHederaSystemContracts(@NonNull final GasCalculator gasCalculator) {
-        return Map.of(Address.fromHexString(PRNG_PRECOMPILE_ADDRESS), new PrngSystemContract(gasCalculator));
+    static Map<Address, HederaSystemContract> provideHederaSystemContracts(
+            @NonNull final HtsSystemContract htsSystemContract, @NonNull final PrngSystemContract prngSystemContract) {
+        return Map.ofEntries(
+                entry(Address.fromHexString(HTS_PRECOMPILE_ADDRESS), requireNonNull(htsSystemContract)),
```

Comment:
Curious if there is a way to provide different maps for `@QueryScope` vs `@TransactionScope`.  The reason I ask is that the `htsSystemContract` looks very different depending on the scope.  Right now in mono service, `HtsPrecompileContract` calls `EvmHTSPrecompiledContract` to handle the view functions if the transaction is static. It might be convenient to just have dagger provide different maps so that one class does not have to call the other.

Reply:
I think that's just an (unfortunate) implementation detail in mono-service.

In the new `HtsSystemContract`, we still need to prevent `static` frames from making state changes no matter if `@QueryScope` or `@TransactionScope`; since all `ContractCallLocal` are static, there shouldn't be any additional special cases.